code/ship/ship.cpp

```diff
@@ -9123,9 +9186,9 @@ static void ship_auto_repair_frame(int shipnum, float frametime)
 
 			// do incremental repair on the subsystem
 			// check for overflow of current_hits
-			ssp->current_hits += ssp->max_hits * real_repair_rate * frametime;
-			if ( ssp->current_hits > ssp->max_hits ) {
-				ssp->current_hits = ssp->max_hits;
+			ssp->current_hits += max_subsys_repair_hits * real_repair_rate * frametime;
+			if (ssp->current_hits > max_subsys_repair_hits) {
+				ssp->current_hits = max_subsys_repair_hits;
```

Comment:
As above, prevents subsystems from being repaired to above the threshold from any source, etc.